File: TerminalHelper-mac/Sources/TerminalHelper/WorkspaceTarget.swift
import Foundation

enum WorkspaceTarget: String, CaseIterable, Equatable, Identifiable, Sendable {
    case terminal
    case visualStudioCode
    case intelliJIdea

    var id: Self { self }

    var displayName: String {
        switch self {
        case .terminal:
            "终端"
        case .visualStudioCode:
            "Visual Studio Code"
        case .intelliJIdea:
            "IntelliJ IDEA"
        }
    }

    var systemImageName: String {
        switch self {
        case .terminal:
            "terminal"
        case .visualStudioCode:
            "chevron.left.forwardslash.chevron.right"
        case .intelliJIdea:
            "hammer"
        }
    }

    var bundleIdentifiers: [String] {
        switch self {
        case .terminal:
            []
        case .visualStudioCode:
            ["com.microsoft.VSCode"]
        case .intelliJIdea:
            ["com.jetbrains.intellij", "com.jetbrains.intellij.ce"]
        }
    }
}

File: TerminalHelper-windows/src/TerminalHelper.Core/Launching/WorkspaceTarget.cs
namespace TerminalHelper.Core.Launching;

public enum WorkspaceTarget
{
    Terminal,
    VisualStudioCode,
    IntelliJIdea,
}

public static class WorkspaceTargetExtensions
{
    public static string GetDisplayName(this WorkspaceTarget target)
    {
        return target switch
        {
            WorkspaceTarget.Terminal => "Terminal",
            WorkspaceTarget.VisualStudioCode => "Visual Studio Code",
            WorkspaceTarget.IntelliJIdea => "IntelliJ IDEA",
            _ => throw new ArgumentOutOfRangeException(nameof(target), target, null),
        };
    }
}
